### ISSUE TITLE
Different behavior between MockHttp 6.0.0 and 7.0.0

### ISSUE THREAD

**Issue by SylvainFI** [richardszalay/mockhttp#143]
Test

```
HttpClient.When("/api/Test/*").Respond("application/pdf", new MemoryStream());
var sut = RenderComponent<MyComponent>(parameters => parameters
    .Add(p => p.Property1, "Test")
    .Add(p => p.Property2, data)
);
```

MyComponent.razor.cs
```
var result = await Http.GetAsync($"api/RouteNotMockedInTest/{Property1}", CancellationTokenSource.Token);
if (!result.IsSuccessStatusCode)
...
```

With &lt;PackageReference Include="RichardSzalay.MockHttp" Version="6.0.0" /&gt; result contains:
```
{StatusCode: 404, ReasonPhrase: 'No matching mock handler for "GET http://localhost/api/RouteNotMockedInTest/Test"', Version: 1.1, Content: System.Net.Http.EmptyContent, Headers:
{
  Content-Length: 0
}}
```

With &lt;PackageReference Include="RichardSzalay.MockHttp" Version="7.0.0" /&gt; line with result is never reached:
Call is swallowed and no error is raised.

**Comment by SylvainFI** [comment]
Ok I added this below to Setup tests. It would be nice to have a better changelog as it was the first time I had to deal with the lib, I didn't know how it was working with version 6.

`HttpClient.Fallback.Respond(HttpStatusCode.NotFound);`

**Comment by richardszalay** [comment]
Hi SylvainFI,

The breaking change (which necessitated the major version change) was outlined in the changeling:

- https://github.com/richardszalay/mockhttp/blob/master/CHANGELOG 
- https://www.nuget.org/packages/RichardSzalay.MockHttp#releasenotes-body-tab

I did consider documenting how to revert the behaviour, but I ended up deciding that it was unlikely that a test would be intentionally passing while relying on the default "unmatched test" behaviour. Clearly a misplaced assumption!

Glad you got it all sorted.
